src/views/Project/Details/Frameworks/index.js

```diff
@@ -64,18 +85,25 @@ const getActiveFrameworkId = memoize((
         }
     }
 
-    let activeFrameworkId;
     if (activeFrameworkIdFromProject) {
-        activeFrameworkId = activeFrameworkIdFromProject;
-    } else {
-        activeFrameworkId = frameworkList.length > 0 ?
-            frameworkList[0].id : undefined;
+        const previouslyActiveFrameworkIndex = frameworkList.findIndex(
+            f => f.id === activeFrameworkIdFromProject,
+        );
+
+        if (previouslyActiveFrameworkIndex !== -1) {
+            return activeFrameworkIdFromProject;
+        }
     }
 
-    return activeFrameworkId;
+    return frameworkList.length > 0 ?
```

Comment:
The above block can be simplified by using a single variable:
`activeFrameworkId = activeFrameworkIdFromState || activeFrameworkIdFromProject`